cmd/ocitools/generate.go

```diff
@@ -301,64 +310,79 @@ func setupSpec(g generate.Generator, context *cli.Context) error {
 		}
 	}
 
-	var sd string
-	var sa, ss []string
-
-	if context.IsSet("seccomp-default") {
-		sd = context.String("seccomp-default")
+	err := addSeccomp(spec, context)
+	if err != nil {
+		return err
 	}
 
-	if context.IsSet("seccomp-arch") {
-		sa = context.StringSlice("seccomp-arch")
-	}
+	return nil
+}
 
-	if context.IsSet("seccomp-syscalls") {
-		ss = context.StringSlice("seccomp-syscalls")
-	}
+func addSeccomp(spec *rspec.Spec, context *cli.Context) error {
+	var secc rspec.Seccomp
 
-	if sd == "" && len(sa) == 0 && len(ss) == 0 {
-		return nil
-	}
+	seccompDefault := context.String("seccomp-default")
+	seccompArch := context.String("seccomp-arch")
+	seccompKill := context.String("seccomp-kill")
+	seccompTrace := context.String("seccomp-trace")
+	seccompErrno := context.String("seccomp-errno")
+	seccompTrap := context.String("seccomp-trap")
+	seccompAllow := context.String("seccomp-allow")
 
 	// Set the DefaultAction of seccomp
-	if context.IsSet("seccomp-default") {
-		if err := g.SetLinuxSeccompDefault(sd); err != nil {
-			return err
-		}
+	if seccompDefault == "" {
```

Comment:
This shouldn't be defaulted to allow. We should figure out intent and error out if we can't. For e.g. if user only passes seccomp-allow, then we can assume user is building a whitelist and we can default to `errno`.